docs/Changelog.md

```diff
@@ -2079,6 +2079,46 @@ opset_import {
 <dd>Constrain input and output types to float tensors.</dd>
 </dl>
 
+### <a name="If-1"></a>**If-1**</a>
+
+  If conditional
+
+#### Versioning
+
+This operator is used if you are using version 1 of the default ONNX operator set until the next BC-breaking change to this operator; e.g., it will be used if your protobuf has:
+
+~~~~
+opset_import {
+  version = 1
+}
+~~~~
+
+#### Attributes
+
+<dl>
+<dt><tt>else_branch</tt> : graph (required)</dt>
+<dd>Graph to run if condition is false</dd>
+<dt><tt>then_branch</tt> : graph (required)</dt>
+<dd>Graph to run if condition is true</dd>
+</dl>
+
+#### Inputs
+
+<dl>
+<dt><tt>cond</tt> : B</dt>
+<dd>Condition for the if</dd>
+</dl>
+
+#### Outputs
+
```

Comment:
This doesn't seem to have rendered correctly; surely there should be some outputs?

Reply:
Rendering is correct, currently there is no outputs spec in the schema.

Reply:
I guess we're going to need to write some custom validation code or generalize the checker, because if the checker rejects any If node that does not has zero outputs, this will not be very useful schema

Reply:
We don't have any way to specify variadic outputs at this time, right?

Reply:
Create an issue?

Reply:
@jamesr66a we "do": https://github.com/onnx/onnx/blob/65b4eb726592a0c847b811163763f248efb129f0/onnx/defs/experiments/defs.cc#L285 But IIUC it's restricted in the sense all outputs have to be of the same type. 

@linkerzhang

Reply:
It definitely doesn't make sense to restrict all outputs to be the same type.

Reply:
@linkerzhang  can you confirm if this is in fact the behavior? Are variadic inputs/outputs constrained to be all the same type?